bundler/lib/bundler/cli/gem.rb

```diff
@@ -308,6 +309,52 @@ def ask_and_set_ci
       ci_template
     end
 
+    def ask_and_set_linter
+      linter_template = options[:linter] || Bundler.settings["gem.linter"]
+      linter_template = deprecated_rubocop_option if linter_template.nil?
+
+      if linter_template.to_s.empty?
+        Bundler.ui.confirm "Do you want to add a code linter and formatter to your gem? " \
+          "Supported Linters:\n" \
+          "* RuboCop:       https://rubocop.org\n" \
+          "* Standard:      https://github.com/testdouble/standard\n" \
+          "\n"
+        Bundler.ui.info hint_text("linter")
+
+        result = Bundler.ui.ask "Enter a linter. rubocop/standard/(none):"
+        if result =~ /rubocop|standard/
+          linter_template = result
+        else
+          linter_template = false
+        end
+      end
+
+      if Bundler.settings["gem.linter"].nil?
+        Bundler.settings.set_global("gem.linter", linter_template)
+      end
+
+      if options[:linter] == Bundler.settings["gem.linter"]
+        Bundler.ui.info "#{options[:linter]} is already configured, ignoring --linter flag."
+      end
+
+      linter_template
+    end
+
+    def deprecated_rubocop_option
+      if !options[:rubocop].nil?
+        if options[:rubocop]
+          Bundler::SharedHelpers.major_deprecation 2, "--rubocop is deprecated, use --linter"
+          "rubocop"
+        else
+          Bundler::SharedHelpers.major_deprecation 2, "--no-rubocop is deprecated, use --linter"
+          false
+        end
+      elsif !Bundler.settings["gem.rubocop"].nil?
+        Bundler::SharedHelpers.major_deprecation 2, "gem.rubocop is deprecated, use gem.linter"
```

Comment:
Here I'd say something like

> Bundler is using your existing `gem.rubocop` configuration, but this configuration is deprecated. Make sure you migrate this configuration by running `bundle config set linter <value>`.

But then you'd need to implement the 6th bullet point I suggesed in https://github.com/rubygems/rubygems/pull/4411#issuecomment-787450937, and maybe I'm asking too much from this PR... :thinking:

Reply:
Ok, I've implemented and tested that case now. If `gem.rubocop` is set, gem.linter will be written and gem.rubocop will be deleted (with a message to that effect). [test here](https://github.com/searls/rubygems/blob/master/bundler/spec/commands/newgem_spec.rb#L1015-L1034)

Reply:
Thanks!

Well my suggestion was recommending running `bundle config gem.linter <value>` in the deprecation message and changing the behaviour of `bundle config` to do the update. I was thinking of suggesting on the fly updating of the old setting like you did here, but I thought it was too magical. But after a second thought your approach is nicer and simpler :+1:.

Reply:
Just to mention, I do remember some reports of running rubygems on read-only file systems complaining about unexpected file writes. But since this config is not meant for production runtime, but for devs, I think it is ok to migrate on the fly. :muscle:

Reply:
Yeah, I had the same thought, but since `bundle gem` already writes configuration and it's never been an issue, this seems fine too indeed!